vehicle_params/easy_driver.py

```diff
@@ -116,35 +116,72 @@ def __init__(self):
         self.rear_lever_arms = pt_i_arr * 0.0254  # in to m
-        
+
         # ~~~ Tires & Pacejka ~~~ #
         # NOTE: These lateral fits all assumed slip angle was in DEGREES, not RADIANS
         self.front_tire_coeff_Fy = [1.69, 0.0004384, 2.769, 614.3, 1496, 0.01784, 0.000432, 0.7237, 0.0001746, 0.1366,
                         -0.1482, -0.06455, 10.45, 3.036e-05, 0.04111, 0.002054, 0.01834, -0.06673]
         self.rear_tire_coeff_Fy = [1.384, -0.0003117, -2.936, 668.1, 1599, 0.03877, 0.0003177, 0.6252, 7.733e-05, -0.08382,
                 -0.1171, 0.04597, 3.107, 5.41e-05, 0.04736, 0.005249, 0.0508, -0.1956]
-        
+
         self.front_tire_spring_coeffs = [624 * 175, 0.5 / 0.0254] # N/m
         self.rear_tire_spring_coeffs = [715.3 * 175, 0.486 / 0.0254] # N/m
-        
+
         # TODO: implement aligning moment & fitting
         #self.front_tire_coeff_Mz = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
         #self.rear_tire_coeff_Mz = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
-        
+
         #TODO: implement accel/braking corner & fitting
         #self.rear_tire_coeff_Fx = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
         #self.front_tire_coeff_Fx = [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]
-        
-        
-        
+
+
+
         ### aerodynamics params ###
-        
-        self.air_temperature = 27 # Celsius
+
+        self.air_temperature = 33.8889 # Celsius
         self.ClA_tot = 4.384
         self.CdA_tot = 1.028
-        self.CsA_tot = 33.91
+        self.CsA_tot = 5.673
         self.CdA0 = 0.7155 # drag coefficient from non aero componenets
         self.static_ride_height = 0.0762 # m
+        self.CsA0 = 8.43
+
+        # distribution of downforce across components
+        ClA_dist = [0.474, 0.289, 0.236]   # [front, undertray, rear]
+        CdA_dist = [0.425, 0.178, 0.396]
+        CsA_dist = [0.666, 0.000, 0.333]
+
+        # pitch, body_slip, and roll sensitivities,
+        #            Cl              Cd               Cs
+        self.p_sens	= [[[.01,   -.06],   [.07,   -.055],  [0,0]], # front [pos, neg] -> [%/deg]
+                      [[.0076, -.0457], [.0546, -.0434],  [0,0]], # undertray
+                      [[.0178, -.0245], [.0294, -.0478], [0,0]]]  # rear
+
+         #                 front               undertray           rear
+        self.bs_sens	= [[.008,  .0114, 0], [.0061,  .0089, 0], [-.0018, -.0058, 0]] # [Cl, Cd, Cs] -> [%/deg]
+        self.r_sens	=     [[-.018,  0,    0], [-.0137,  0,    0], [-.005,  -.0145, 0]]
+
+         # conversion factors
+        self.in_to_m = 0.0254
+        self.rad_to_deg = 180 / np.pi
+
+        # positions of component CoPs (magnitudes, equation takes signs into account)
+        # TODO: make positions relative to intermediate frame; these lines also seem wrong in general ATM
+        # Front, Undertray and Rear [x , y , z]
+        self.CoP = np.array([[23.65 * self.in_to_m + self.cg_total_position[0],  0, 9.30 * self.in_to_m],
```

Comment:
Can we have this array be purely CoP location values from CAD measured in inches? So bring the calculation and CG offsets into the Aerodynamics file. I remember you started it during the workday but I may have gotten lost after all the vectorizing.

Reply:
Fixed in latest commit.